src/jit/codegenarmarch.cpp

```diff
@@ -1554,11 +1554,13 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     emitAttr    attr       = emitTypeSize(tree);
     instruction ins        = ins_Load(targetType);
 
-    assert((attr != EA_1BYTE) || !(tree->gtFlags & GTF_IND_UNALIGNED));
-
     genConsumeAddress(tree->Addr());
     if (tree->gtFlags & GTF_IND_VOLATILE)
```

Comment:
Since you're in this code, it would be nice to fix this: `if ((tree->gtFlags & GTF_IND_VOLATILE) != 0)`